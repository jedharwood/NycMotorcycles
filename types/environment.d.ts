declare global {
  namespace NodeJS {
    interface ProcessEnv {
      NEXT_PUBLIC_API_MOCKING: string;
      SELECTOR_PRODUCT: string;
      SELECTOR_PRODUCT_PRICE: string;
      SELECTOR_PRODUCT_TIME: string;
      SELECTOR_PRODUCT_TITLE: string;
      SELECTOR_PRODUCT_BID: string;
      YAHOO_AUCTION_PROFILE_PAGE_URL: string;
      MAILGUN_API_KEY: string;
      MAILGUN_DOMAIN: string;
      DELIVERY_EMAIL: string;
    }
  }
}

export {};
