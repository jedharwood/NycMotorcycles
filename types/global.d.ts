type TimeRemaining = {
  time: number | undefined;
  unit: 'days' | 'hours' | 'minutes';
};

type ActiveAuction = {
  title?: string;
  url?: string;
  image: GridImage;
  bidders?: string;
  timeRemaining?: TimeRemaining;
  currentPrice?: string;
  promptDecisionPrice?: string;
};

type GridImage = {
  imageSrc: string;
  imageAlt: string;
  width?: number;
  height?: number;
  onImageClick?: () => void;
};

type ArchiveBikeKey = keyof typeof archive;

type ContactFormData = {
  email: string;
  senderName: string;
  subject: string;
  message: string;
};

type InputFieldProps = {
  type: 'email' | 'text' | 'text-area';
  placeholder: string;
  name: 'email' | 'senderName' | 'subject' | 'message';
  register: UseFormRegister<ContactFormData>;
  error: FieldError | undefined;
  label: string;
};

type ComponentColour = 'green' | 'red';

type SoldOrCall = 'sold' | 'call';

type TheNumbersOneToFour = 1 | 2 | 3 | 4;

type NavLinkDetails = {
  href: string;
  text: string;
};

type FooterProps = {
  address: string;
  disclaimer: string;
  chevronText: string;
  altTextInstagramButton: string;
};

type ChildElementPosition = 'under-title' | 'bottom';

type TextDisplayProps = {
  title?: string;
  textContent?: string[];
  childElement?: JSX.Element;
  childElementPosition?: ChildElementPosition;
  isOpaque?: boolean;
  hasBorder?: boolean;
  textContentCentred?: boolean;
  borderColour?: ComponentColour;
  subTitle?: string;
  footer?: string;
};