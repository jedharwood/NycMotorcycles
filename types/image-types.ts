import { StaticImageData } from 'next/image';

export type StaticImage = {
  image: StaticImageData;
  altText: string;
  onImageClick?: () => void;
};

export type ImageIndex = {
  [key: string]: StaticImage;
};

export const blankStaticImage: StaticImage = {
  image: {
    src: '',
    height: 0,
    width: 0,
  },
  altText: '',
};