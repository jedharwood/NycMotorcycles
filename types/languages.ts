import en from '../languages/en.json';
import ja from '../languages/ja.json';

export const langs = {
    en: 'en',
    ja: 'ja',
};

type Messages = {
    [key: string]: {
        [key: string]: string;
    };
}; 

export const messages: Messages = {
    en,
    ja,
};