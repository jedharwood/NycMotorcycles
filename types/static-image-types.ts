import { StaticImageData } from 'next/image'

export type StaticImage = {
    image: StaticImageData;
    altText: string;
    onImageClick?: () => void;
}

export type StaticImageCatalog = {
    [key: string]: StaticImage;
  };