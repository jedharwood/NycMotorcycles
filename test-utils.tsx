import React, { ReactElement } from 'react';

import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';

import { langs, messages } from './types/languages';

type ProviderProps = {
    children: React.ReactNode;
    locale: string;
};

const AllTheProviders = ({ children, locale }: ProviderProps): JSX.Element => {
    const queryClient = new QueryClient();

    return (
        <MemoryRouterProvider>
            <QueryClientProvider client={queryClient}>
                <IntlProvider locale={locale} messages={messages[locale]}>
                    {children}
                </IntlProvider>
            </QueryClientProvider>
        </MemoryRouterProvider>
    );
};

type RenderWithProvidersOptions = RenderOptions & {
    locale?: string;
};

const renderWithProviders = (
    ui: ReactElement,
    { locale = langs.en, ...renderOptions }: RenderWithProvidersOptions = {}
) =>
    render(ui, {
        wrapper: ({ children }) => (
            <AllTheProviders locale={locale}>{children}</AllTheProviders>
        ),
        ...renderOptions,
    });

export * from '@testing-library/react';
export { renderWithProviders as render };
