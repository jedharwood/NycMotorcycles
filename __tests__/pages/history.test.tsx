import userEvent from '@testing-library/user-event';
import mockRouter from 'next-router-mock';

import HistoryPage from '@/pages/history';
import { render, screen, waitFor, act } from '@/test-utils';
import { langs } from '@/types/languages';

jest.mock('next/router', () => jest.requireActual('next-router-mock'));

describe('HistoryPage', () => {
    beforeEach(() => {
        mockRouter.setCurrentUrl('/history');
    });

    afterEach(() => {
        jest.restoreAllMocks();
    });

    it('should render page in English', () => {
        const { container } = render(<HistoryPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render page in Japanese', () => {
        const { container } = render(<HistoryPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });

    it('should navigate to home page when Home button is clicked', async () => {
        const user = userEvent.setup();
        render(<HistoryPage />);

        await expect(mockRouter.asPath).toEqual('/history');

        const homeButton: HTMLElement = screen.getByTestId('history-page-home-button');

        await act(() => {
            userEvent.click(homeButton);
        });

        await waitFor(() => {
            expect(mockRouter.asPath).toEqual('/');
        });
    });
});
