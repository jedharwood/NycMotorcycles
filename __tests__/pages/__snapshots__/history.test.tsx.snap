// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryPage should render page in English 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="A black and white image of Larry from New York City Motorcycles posing on a vintage motorcycle wearing a fur hat. He is wearing sunglasses and rider's goggles; it's a wonder he can see anything at all."
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            History
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-5"
      >
        <div
          class="md:col-span-3"
        >
          <section
            class="grid gap-4 sm:grid-cols-2 md:grid-cols-2"
          >
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="A crowd of vintage motorbike enthusiasts gathered outside the New York City Motorcycles showroom to watch a band that is playing inside."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An image of a David Karp, the founder of Tumblr leaning on an orange coloured Laverda motorbike outside the New York City Motorcycles showroom."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An image of Larry from New York City Motorcycles performing a last minute tune-up on his bike wearing full race leathers."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An orange Laverda and a red Ducati vintage motorbikes basking in the sunlight in the pit on race day."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An image of Larry from New York City Motorcycles leaning deep into a turn on the race track riding an orange coloured Laverda motorbike."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="A black and white image of a motorbike being lifted by a forklift truck outside the New York City Motorcycles showroom."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An image of the New York City Motorcycles garage in Venice, California. There are a lot of vintage motorbikes crowded in there; most notably a lovely old BSA motorcycle."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="The New York City Motorcycles garage viewed from outside."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="A satisfied customer riding his new motorbike away from New York City Motorcycles HQ in Zushi, Kanagawa, Japan."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An image of a shipping container full of vintage cars and motorcycles on their way to Japan."
              />
            </div>
          </section>
        </div>
        <div
          class="space-y-6 md:col-span-2"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2012
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                New York City Motorcycle "club" starts in a Soho loft in Manhattan. The idea is simple: to givr an identity and a home base to NYC's unique vintage motorcycle scene. Riding and owning motorcycles in New York City is cumbersome and complicated. The payoff is experiencing the greatest city in the world on two wheels. From toughness to style, riding bikes in New York is not like anywhere else. New York City Motorcycles provided a platform for like-minded enthusiasts enduring the pain and pleasure of motorcycling in "The Big Apple".
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2014
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                New York City Motorcycles starts AHRMA Roadracing, supported by buying and selling vintage motorcycles. Race school at Roebling Road Raceway, Pooler Georgia, pictured.
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2015
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                Brooklyn showroom and retail shop opens in the center of New York's legendary vintage and custom "Motorcycle District" on North 14th Street in Williamsburg. N14th is home to Indian Larry, Works Engineering, The Brooklyn Invitational and NYCVINMOTO motorcycle shows.
              </p>
              <p>
                Racing program expands to three bikes with coast-to-coast races on a Laverda 750SFC race replica, Ducati 250 and Triumph 500.
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2016
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                New York City Motorcycles opens a garage in Venice, California, adjacent to Deus Ex Machina.
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2017
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                Following a major race crash ans the sale of the building, NYCMC's Brooklyn shop closes. Full relocation to Venice, CA completed. Resumes racing, including first international competition at the 5oth anniversary of the first Canadian GP at Mosport.
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2018
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                Most of the year is spent on the road, lving in an Airstream and racing, buying and selling vintage motorcycles. A most blissful year.
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2019
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                Get's married, moves to Japan.
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                2020
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                New York City Motorcycles opens in Kanagawa, on Japan's famous Shonan coast, sourcing and exporting motorcycles worldwide.
              </p>
            </div>
            <div
              class="flex justify-center"
            >
              <a
                data-testid="history-page-home-button"
                href="/"
              >
                <div
                  class="flex justify-center"
                >
                  <button
                    class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
                    type="button"
                  >
                    Home
                  </button>
                </div>
              </a>
            </div>
          </section>
        </div>
        <div
          class="md:col-span-5"
        >
          <section
            class="grid gap-4 sm:grid-cols-2 md:grid-cols-2"
          >
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An image of a motorbike passing customs checks at a Japanese port."
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="An exterior shot of a vintage motorcycle being loaded onto the back of a small flatbed truck. There's a large ferris wheel in the background so it might be in Yokohama."
              />
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`HistoryPage should render page in Japanese 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="placeholder"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            placeholder
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-5"
      >
        <div
          class="md:col-span-3"
        >
          <section
            class="grid gap-4 sm:grid-cols-2 md:grid-cols-2"
          >
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
          </section>
        </div>
        <div
          class="space-y-6 md:col-span-2"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                placeholder
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                placeholder
              </p>
            </div>
            <div
              class="flex justify-center"
            >
              <a
                data-testid="history-page-home-button"
                href="/"
              >
                <div
                  class="flex justify-center"
                >
                  <button
                    class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
                    type="button"
                  >
                    placeholder
                  </button>
                </div>
              </a>
            </div>
          </section>
        </div>
        <div
          class="md:col-span-5"
        >
          <section
            class="grid gap-4 sm:grid-cols-2 md:grid-cols-2"
          >
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
            <div
              class="rounded-md
            card-hover cursor-pointer 
            "
            >
              <img
                alt="placeholder"
              />
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
